doc/en/assert.txt

```diff
@@ -113,6 +113,38 @@ exceptions your own code is deliberately raising, whereas using
 like documenting unfixed bugs (where the test describes what "should" happen)
 or bugs in dependencies.
 
+Assertions about expected warnings
+-----------------------------------------
+
+.. versionadded:: 2.8
+
+You can also check that code raises a particular warning using ``pytest.warns``,
+which works in a similar manner to ``pytest.raises`` (see above)::
+
+    import warnings
+    import pytest
+
+    def test_warning():
+        with pytest.warns(UserWarning):
+            warnings.warn("my warning", UserWarning)
+
+The test will fail if the warning in question is not raised. The function also
+returns a list of all raised warnings (as ``warnings.WarningMessage`` objects),
+which you can query for additional information::
+
+    with pytest.warns(RuntimeWarning) as record:
+        warnings.warn("another warning", RuntimeWarning)
+
+    assert len(record) == 1  # check that only one warning was raised
+    assert record[0].message.message == "another warning"
+
+Alternatively, you can examine raised warnings in detail using the ``recwarns``
```

Comment:
I think this paragraph should be moved to after the next one (`"You can also call..."`), plus it would be nice to contain a link to the `recwarns` docs. :smile: